src/odemis/gui/comp/overlay/shapes.py

```diff
@@ -45,6 +47,11 @@ def __init__(self, cnvs):
         # Useful for internal points manipulation
         self._points: List[Vec] = []
         self.cnvs = cnvs
+        # History of states
+        self.undo_stack = deque()
+        self.redo_stack = deque()
```

Comment:
If you use deque, then you should `max_len` to automatically discard the oldest elements. Otherwise, using a list is as simple.